test/prism/PrismLogicTest.scala

```diff
@@ -330,4 +330,51 @@ class PrismLogicTest extends FreeSpec with Matchers {
       }
     }
   }
-}
\ No newline at end of file
+
+  "sortInstancesByStack" - {
+    "should sort by stack, app, stage" - {
+      "when stack and stage are non empty" in {
+        val in1 = Instance("arn", "name", "state", "group", "dns","ip", DateTime.now(), "instanceName", "region", "vendor", List.empty,Map.empty, Some("stack1"), Some("stageB"), List("appB"), List.empty, Map.empty, null)
+        val in2 = in1.copy(stack = Some("stack2"), stage = Some("stageA"), app = List("appA"))
+        val instanceList = List(in2, in1)
+        PrismLogic.sortInstancesByStack(instanceList) should contain inOrderOnly(in1, in2)
+      }
+      "when stack is empty" in {
+        val in1 = Instance("arn", "name", "state", "group", "dns","ip", DateTime.now(), "instanceName", "region", "vendor", List.empty,Map.empty, None, Some("stageB"), List("app1"), List.empty, Map.empty, null)
+        val in2 = in1.copy(stage = Some("stageA"), app = List("app2"))
+        val instanceList = List(in2, in1)
+        PrismLogic.sortInstancesByStack(instanceList) should contain inOrderOnly(in1, in2)
+      }
+      "when stack and stage are empty" in {
+        val in1 = Instance("arn", "name", "state", "group", "dns","ip", DateTime.now(), "instanceName", "region", "vendor", List.empty,Map.empty, None, None, List("app1"), List.empty, Map.empty, null)
+        val in2 = in1.copy(app = List("app2"))
+        val instanceList = List(in2, in1)
+        PrismLogic.sortInstancesByStack(instanceList) should contain inOrderOnly(in1, in2)
+      }
+    }
+  }
+
+  "sortLCsByOwner" - {
+    "should sort Launch Configurations" - {
```

Comment:
Again, this nesting doesn't achieve much. I'll generally only introduce a block where there's a reason for doing so, for example to introduce some block-scoped test data or as a helpful description of a group of tests.